### PR TITLE
Serializer cleanup

### PR DESCRIPTION
**Pull request by srh**
- [x] I have read and agreed to the RethinkDB Contributor License Agreement http://rethinkdb.com/community/cla/

### Description
In making changes to the serializer, I put all the cleanup and janitorial commits up front.  So this is that set of commits.

I guess the commits fall into 5 categories:
- Non sequitur commits, don't really have much to do with the serializer
- Further cleanup as a result of the prior removal of the semantic checking serializer
- Moving stuff around so that changes can be tested or otherwise implemented
- Getting rid of all the silly little metablock typedefs
- Simplying some absolutely crazy way of doing stuff (e.g. metablock_manager_t::head_t)

### CODE REVIEW

File: src/rdb_protocol/changefeed.cc [hunk 1]
```diff
@@ -1572,8 +1570,8 @@ class flat_sub_t : public subscription_t {
             queue->add(change_val_t(
                 std::make_pair(shard_uuid, stamp),
                 pkey,
-                old_val,
-                new_val
+                std::move(old_val),
+                std::move(new_val)
```
Comment (AtnNn) [inline src/rdb_protocol/changefeed.cc]: These two parameters are still being copied.

Comment (srh) [inline src/rdb_protocol/changefeed.cc]: This is now fixed -- the type is noncopyable now.  (And I think it was actually being moved before?)

**Comment by AtnNn** [comment]
LGTM

**Comment by AtnNn** [comment]
Pushed to`next` (68ea014)

**Pull request closed without merging**